apps/frontend/src/libs/components/select/libs/helpers/select/select.helper.ts

```diff
@@ -0,0 +1,24 @@
+import { type MultiValue, type SingleValue } from "react-select";
+
+import { type SelectOption } from "../../types/types.js";
+import { isMultiValue } from "../helpers.js";
+
+const createHandleChange =
+	<TOptionValue>(
+		onChange: (value: null | TOptionValue | TOptionValue[]) => void,
+	) =>
+	(
+		newValue:
+			| MultiValue<SelectOption<TOptionValue>>
+			| SingleValue<SelectOption<TOptionValue>>,
+	): void => {
+		if (isMultiValue<TOptionValue>(newValue)) {
+			const values = newValue.map((o) => o.value);
+			onChange(values);
+		} else {
+			const single = newValue?.value ?? null;
+			onChange(single);
+		}
+	};
```

Comment:
It's better to simplify logic here to return a value, no need to call the helper itself
```suggestion
const getSelectNewValue =
	<TOptionValue>(
		onChange: (value: null | TOptionValue | TOptionValue[]) => void,
	) => MultiValue<SelectOption<TOptionValue>>
			| SingleValue<SelectOption<TOptionValue>>=> {
	   return Array.isArray(newValue) ? newValue.map((option) => option.value) : newValue?.value ?? null;
	};
```